src/packages/external-event/external-event.ts

```diff
@@ -0,0 +1,136 @@
+import type { Express, Request, Response } from 'express';
+import type { ExternalEventTypeInsertInput } from '../../types/external-event.js';
+import Ajv from 'ajv';
+import { getEnv } from '../../env.js';
+import getLogger from '../../logger.js';
+import gql from './gql.js';
+import { HasuraError } from '../../types/hasura.js';
+
+const logger = getLogger('packages/external-event/external-event');
+const { HASURA_API_URL } = getEnv();
+const GQL_API_URL = `${HASURA_API_URL}/v1/graphql`;
+
+const ajv = new Ajv();
+
+async function uploadExternalEventType(req: Request, res: Response) {
+  logger.info(`POST /uploadExternalEventType: Entering function...`);
+  const authorizationHeader = req.get('authorization');
+  logger.info(`POST /uploadExternalEventType: ${authorizationHeader}`);
+
+  const {
+    headers: { 'x-hasura-role': roleHeader, 'x-hasura-user-id': userHeader },
+  } = req;
+
+  const { body } = req;
+  const { external_event_type_name, attribute_schema } = body;
+
+  const headers: HeadersInit = {
+    Authorization: authorizationHeader ?? '',
+    'Content-Type': 'application/json',
+    'x-hasura-role': roleHeader ? `${roleHeader}` : '',
+    'x-hasura-user-id': userHeader ? `${userHeader}` : '',
+  };
+
+  // Validate schema is valid JSON Schema
+  try {
+    const schemaIsValid: boolean = ajv.validateSchema(attribute_schema);
+    if (!schemaIsValid) {
+      throw new Error("Schema was not a valid JSON Schema.");
+    }
+  } catch (error) {
+    logger.error((error as Error).message);
+    res.status(500);
+    res.send((error as Error).message);
```

Comment:
Can `ajv.validateSchema` throw an Error? If not, why is this a try-catch? If it can, then this comment can be resolved.

Reply:
Good catch - no it looks like all errors are just directly accessible through `ajv.errors`. I updated the `logger` and `res.send` statements to use `ajv.errors` instead of relying on a `try / catch`.